pkg/api/dashboard.go

```diff
@@ -361,6 +362,15 @@ func (hs *HTTPServer) postDashboard(c *models.ReqContext, cmd models.SaveDashboa
 
 	dashboard, err := hs.dashboardService.SaveDashboard(alerting.WithUAEnabled(ctx, hs.Cfg.UnifiedAlerting.IsEnabled()), dashItem, allowUiUpdate)
 
+	if dashboard != nil && hs.entityEventsService != nil {
```

Comment:
Sounds like we only need this because we don't have any communication between grafana servers in HA?

Reply:
Exactly - the alternative is to use different implementations depending on the the type of environment, e.g. in-mem messaging in non-HA and sql-based "pubsub" in HA

My assumption is that SQL-based solution won't put too much extra strain on the DB as we are only tracking changes (rather than changes+reads), so I think we can start with a single implementation across all envs and then reevaluate after observing its performance during tests

Reply:
@bergquist -- correct, this structure is so things can work with HA.  We discussed other options but all require more tools (redis, jetstream, etc).  In this PR we are trying to get something basic working with the current infrastructure; we can then explore other options.